### PR TITLE
Fix wrong pointer passed to Fortran MPI in FFTE

### LINKED ISSUE #13: FFTE backend
Currently, we need to address:
- Modify CMake for correct compilation, using GCC 7 or higher. In case PGI is needed, we need to put that option accordingly.
- Once compiling, use the test3D_CPU_C2C.c    and    test3D_GPU_C2C.c , for verifying if routines from FFTE are being correctly called.  The Error checker can be deactivated at this point (comment that section out).
- If possible, verify correctness by printing out the output of the forward FFTE and compare to the one from FFTW or heFFTe backends.

### PR DESCRIPTION
**Pull request by luszczek**
This fixes the outstanding problem from #13 and shows how to call other functions from FFTE backend implementation.

### CODE REVIEW

**Comment by luszczek** [comment]
CI seems to be saying something different than my build. I've run the FFTE test with the same set of packages and got no issues. It seems that the build inside CI doesn't get the latest patch and runs the old code that segfaulted before. Here's my output:
```
mpirun -np 2 test3D_C2C -lib ffte -backend fftw -size 4 4 4 -pgrid 1 2
-----------------------------------------------
Benchmarking ffte library using fftw backend
-----------------------------------------------
[0] proc_grid 1x1x2 - [New MPI process 0] I am located at (0, 0).   0-0-0    3-3-1   local_size={32}
[ffte] successfully initialized.
[1] proc_grid 1x1x2 - [New MPI process 1] I am located at (0, 1).   0-0-2    3-3-3   local_size={32}
Benchmarking FFTE: Get Complex-to-Complex backend using ptest3d.f
Benchmarking FFTE: Get Complex-to-Complex backend using ptest3d.f
Time for FFT plan  = 2.3956e-05
Time for execution =    333.864
Benchmarking FFTE: Get Complex-to-Complex backend using ptest3d.f
Benchmarking FFTE: Get Complex-to-Complex backend using ptest3d.f
FFTE: |X - ifft(fft(X)) |_{max}: 3.051562e+01
```

**Comment by G-Ragghianti** [comment]
You are correct.  The CI uninstalls the fiber package at the begining of the job, but doesn't touch the previously installed dependencies.  I will update the CI so that it will uninstall all the FFT libraries as well.  

> CI seems to be saying something different than my build. I've run the FFTE test with the same set of packages and got no issues. It seems that the build inside CI doesn't get the latest patch and runs the old code that segfaulted before. Here's my output:

**Comment by G-Ragghianti** [comment]
Correction: The CI is _supposed_ to be uninstalling the fiber and fft packages, but it doesn't appear to be doing so.  I'm investigating the cause.

File: external/ffte/tools/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+cmake_minimum_required(VERSION 3.8)
```
Comment (af-ayala) [inline external/ffte/tools/CMakeLists.txt]: For uniformity, I would suggest to keep the same 3.10 version as in the main CMakeList?

**Comment by G-Ragghianti** [comment]
OK, the CI is now able to uninstall fiber and the fft library before attempting to install and test. I will run the CI checks on this PR.

**Comment by G-Ragghianti** [comment]
Can you merge the changes that I recently made to the CI into your branch in order for the test to run with the updated CI config?  The changes are under .github/

File: external/ffte/tools/CMakeLists.txt (hunk 1 above)
Comment (luszczek) [reply external/ffte/tools/CMakeLists.txt]: This is not for FIBER. This is for those who build FFTE with CMake. FFTE provides only Makefile's for their tests and nothing else. FFTE doesn't need a new CMake version than 3.8 that has features for libraries and for CUDA language.

Comment (luszczek) [reply external/ffte/tools/CMakeLists.txt]: I pushed d5d4896 to make them uniform

**Comment by luszczek** [comment]
It took me forever to merge your CI branch. I hope it was worth it.

**Comment by luszczek** [comment]
I'd be okay to close this and start from scratch after the tests finish.

**Comment by luszczek** [comment]
Closing due to wrong `rebase`

**Pull request closed without merging**